src/services/TaskRepository.ts

```diff
@@ -1,83 +1,86 @@
-        {
-            queryString += `WHERE completed = ${completion} `;
+    async getTasks(completion:string, sortBy:string = 'dueDate', sortOrder:string = 'asc'){
 
-        }
+        const query = Task.TaskModel()
+        var results
 
-        if (sortBy != null && sortBy != undefined)
-        {
-            queryString += `ORDER BY ${sortBy} ${sortOrder}`;
+        // if completion parameter isn't given, just give all tasks.
+        if (completion != null || completion != undefined){
+            results = query.findAll({
+                where: {
+                    completed : JSON.parse(completion),
+                },
+                order: [
+                    [sortBy , sortOrder]
+                ]
+            })
+        }else{
+            results = query.findAll({
+                order: [
+                    [sortBy , sortOrder]
+                ]
+            })
         }
-        
-        var [results] = await instances.dbs.ToDoAPI.sequelize.query(
-            queryString
-        )
-        return results;
+
+        return results
     }
 
-    async fetchTask(id:number) {
-        const [results] = await instances.dbs.ToDoAPI.sequelize.query(
-            'SELECT * FROM Tasks WHERE id = ?',
-            {
-                replacements: [id],
-                type: QueryTypes.SELECT
+    async fetchTask(fetchID:number) {
+        const query = Task.TaskModel()
+
+        const results = query.findAll({
+            where: {
+                id: fetchID
             }
-        )
-        return results;
+        })
+
+        return results
     }
 
     async createTask(desc:string, due:Date, completed: boolean){
-        const newTask:ITask = {
-            id: 0,
+
+        const query = Task.TaskModel()
+
+        const newTask = await query.create({
+            id: uuidv4(),
             taskDescription:desc, 
-            createdAt: new Date(),
             dueDate:due, 
             completed:completed
-        }
-        const created = newTask.createdAt.toISOString().split('T')[0]
-
-        const [results] = await instances.dbs.ToDoAPI.sequelize.query(
-            'INSERT INTO Tasks (taskDescription, createdAt, dueDate, completed) '+
-            `VALUES (?, ?, ?, ?)`,
-            {
-                replacements:[desc, created, due, completed],
-                type: QueryTypes.INSERT
-            }
-        )
-        return results;
+        })
+        return newTask
     }
 
-    async updateTask(id:number, desc:string, due:Date, completed: boolean) {
-        const [results] = await instances.dbs.ToDoAPI.sequelize.query(
-            'UPDATE Tasks SET ' +
-            `taskDescription = ?, ` +
-            `dueDate = ?, ` +
-            `completed = ? ` +
-            `WHERE id = ?`,
-            {
-                replacements: [desc, due, completed, id],
-                type: QueryTypes.UPDATE
+    async updateTask(fetchID:number, desc:string, due:Date, completed: boolean) {
```

Comment:
How are you handling declaring this function if only a new due date or only a new (description and completed status) are being passed through the payload in your endpoints

Reply:
The ID is gotten from the params rather than the payload